rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> tuple[list[Trade], tuple[Timestamp, Timestamp]]:
+        """
+        Query trades from bybit. This endpoint in the api allows to query by time range
+        """
+        new_trades = []
+        upper_ts, lower_ts = end_ts, Timestamp(end_ts - WEEK_IN_SECONDS)
+        two_years_ago = Timestamp(ts_now() - DAY_IN_SECONDS * 365 * 2)
+        if start_ts < two_years_ago:
+            # bybit allows to query up to two years into the past
+            start_ts = two_years_ago
+
+        while True:
+            options = {
+                'category': 'spot',
+                'endTime': str(ts_sec_to_ms(upper_ts)),
+                'limit': PAGINATION_LIMIT,
+                'startTime': str(ts_sec_to_ms(lower_ts)),
+            }
+            log.debug(f'querying online trades for {options}')
+            raw_data = self._paginated_api_query(endpoint='order/history', options=options)
+            for raw_trade in raw_data:
+                if raw_trade['orderStatus'] != 'Filled':
+                    # the api doesn't allow to filter by status in the classic spot
+                    continue
+
+                try:
+                    base_asset, quote_asset = bybit_symbol_to_base_quote(raw_trade['symbol'])
+                except UnknownAsset:
+                    log.error(f'Could not read assets from trade pair {raw_trade["symbol"]}')
+                    continue
+
+                try:
+                    if raw_trade['orderType'] == 'Market':
+                        rate = deserialize_price(raw_trade['avgPrice'])
+                    else:
+                        rate = deserialize_price(raw_trade['price'])
+                except DeserializationError as e:
+                    log.error(f'{e} when reading rate for bybit trade {raw_trade}')
+                    continue
+
+                try:
+                    trade = Trade(
+                        timestamp=ts_ms_to_sec(TimestampMS(int(raw_trade['updatedTime']))),
+                        location=Location.BYBIT,
+                        base_asset=base_asset,
+                        quote_asset=quote_asset,
+                        trade_type=TradeType.deserialize(raw_trade['side']),
+                        amount=deserialize_asset_amount(raw_trade['qty']),
+                        rate=rate,
+                        fee=deserialize_fee(raw_trade['cumExecFee']) if len(raw_trade['cumExecFee']) else Fee(ZERO),  # noqa: E501
+                        link=raw_trade['orderLinkId'],
+                    )
+                except DeserializationError as e:
+                    log.error(f'{e} when reading rate for bybit trade {raw_trade}')
+                    continue
+
+                new_trades.append(trade)
+
+            upper_ts = Timestamp(upper_ts - WEEK_IN_SECONDS)
+            lower_ts = Timestamp(lower_ts - WEEK_IN_SECONDS)
+
+            if upper_ts <= start_ts:
+                break
```

Comment:
Don't leave so many empty newlines. Explain what you are doing here. The reader can't understand what it is you are doing this for, why a week? What are these upper and lower ts?

```suggestion
            upper_ts = Timestamp(upper_ts - WEEK_IN_SECONDS)
            lower_ts = Timestamp(lower_ts - WEEK_IN_SECONDS)
            if upper_ts <= start_ts:
                break
```